### DISCUSSION TITLE
Conflict Resolution?

### DISCUSSION THREAD

**Discussion by Blankeos** [maxnowack/signaldb#786] [Q&A] [answered]
Noticed this keyword `Conflict Resolution` wasn't mentioned anywhere in the docs. I read a bit more on how replication works here: https://signaldb.js.org/replication/http/

Just wondering what would happen if theoretically in this code snippet. 1 call of `https://api.example.com/todos` threw some errors and did not push successfully. Is the practice with SignalDB to just manually handle this?

```ts
// ...
push: async (changes) => {
      ...changes.added.map(async (item) => {
        await fetch('https://api.example.com/todos', {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            id: item.id,
            text: item.text,
            completed: item.completed,
          }),
        })
      }),
},
// ...
```

**Comment by maxnowack** [comment]
Right now there is no "automatic" conflict resolution or error handling happing inside of the `ReplicatedCollection`. You have to implement it by yourself in the `pull` and `push` handlers.
Since this would be very useful to have directly in the `ReplicatedCollection`, I'm happy to integrate it. 
The specific implementation depends heavily on the use cases. Could you elaborate a bit what you have in mind?

**Reply by Blankeos** [reply]
Thanks for the quick response. I'm actually okay with handling them manually. Just thought those limitations could be noted in the docs. Don't really have ideas, more like just looking for tips on how to handle them correctly (what should be done and not done).

I noticed WatermelonDB's docs address these limitations a bit: 
- https://watermelondb.dev/docs/Sync/FAQ
- https://watermelondb.dev/docs/Sync/Limitations

**Reply by maxnowack** [reply]
You're right. It should be mentioned in the docs. I've added a small hint in https://github.com/maxnowack/signaldb/commit/ca43a81acd05ca884b2c7bc6fb2a071db8c85d7e

**Comment by maxnowack** [comment, marked as answer]
The [`SyncManager`](https://signaldb.js.org/sync/) has built-in conflict resolution.
